precompiles/xcm-transactor/src/lib.rs

```diff
@@ -356,8 +580,72 @@ where
 		let origin = Runtime::AddressMapping::into_account_id(handle.context().caller);
 		let call = pallet_xcm_transactor::Call::<Runtime>::transact_through_signed {
 			dest: Box::new(xcm::VersionedMultiLocation::V1(dest)),
-			fee_currency_id: currency_id,
-			dest_weight: weight,
+			fee: CurrencyPayment {
+				currency: Currency::AsCurrencyId(currency_id),
+				fee_amount: None,
+			},
+			weight_info: TransactWeights {
+				transact_require_weight_at_most: weight,
+				overall_weight: None,
+			},
+			call,
+		};
+
+		RuntimeHelper::<Runtime>::try_dispatch(handle, Some(origin).into(), call)?;
+
+		Ok(succeed([]))
+	}
+
+	fn transact_through_signed_custom_fee_and_weight(
+		handle: &mut impl PrecompileHandle,
+	) -> EvmResult<PrecompileOutput> {
+		handle.record_cost(1 * RuntimeHelper::<Runtime>::db_read_gas_cost())?;
```

Comment:
Maybe we should record the cost right before the actual read?

Reply:
right